tasks/msi.py

```diff
@@ -433,3 +436,85 @@ def MsiClosing(obj):
         yield obj
     finally:
         obj.Close()
+
+
+def get_msm_info(ctx, release_version):
+    """
+    Get the merge module info from the release.json for the given release_version
+    """
+    env = load_release_versions(ctx, release_version)
+    base_url = "https://s3.amazonaws.com/dd-windowsfilter/builds"
+    msm_info = {}
+    if 'WINDOWS_DDNPM_VERSION' in env:
+        info = {
+            'filename': 'DDNPM.msm',
+            'build': env['WINDOWS_DDNPM_DRIVER'],
+            'version': env['WINDOWS_DDNPM_VERSION'],
+            'shasum': env['WINDOWS_DDNPM_SHASUM'],
+        }
+        info['url'] = f"{base_url}/{info['build']}/ddnpminstall-{info['version']}.msm"
+        msm_info['DDNPM'] = info
+    if 'WINDOWS_DDPROCMON_VERSION' in env:
+        info = {
+            'filename': 'DDPROCMON.msm',
+            'build': env['WINDOWS_DDPROCMON_DRIVER'],
+            'version': env['WINDOWS_DDPROCMON_VERSION'],
+            'shasum': env['WINDOWS_DDPROCMON_SHASUM'],
+        }
+        info['url'] = f"{base_url}/{info['build']}/ddprocmoninstall-{info['version']}.msm"
+        msm_info['DDPROCMON'] = info
+    if 'WINDOWS_APMINJECT_VERSION' in env:
+        info = {
+            'filename': 'ddapminstall.msm',
+            'build': env['WINDOWS_APMINJECT_MODULE'],
+            'version': env['WINDOWS_APMINJECT_VERSION'],
+            'shasum': env['WINDOWS_APMINJECT_SHASUM'],
+        }
+        info['url'] = f"{base_url}/{info['build']}/ddapminstall-{info['version']}.msm"
+        msm_info['APMINJECT'] = info
+    return msm_info
+
+
+@task(
+    iterable=['drivers'],
+    help={
+        'drivers': 'List of drivers to fetch (default: DDNPM, DDPROCMON, APMINJECT)',
+        'release_version': 'Release version to fetch drivers from (default: nightly-a7)',
+    },
+)
+def fetch_driver_msm(ctx, drivers=None, release_version=None):
+    """
+    Fetch the driver merge modules (.msm) that are consumed by the Agent MSI.
+
+    Defaults to the versions provided in the @release_version section of release.json
+    """
+    ALLOWED_DRIVERS = ['DDNPM', 'DDPROCMON', 'APMINJECT']
```

Comment:
I had something like that at first, since it was easy with `DDNPM` -> `DDNPM.msm` and `DDPROCMON` -> `DDPROCMON.msm`.
But when adding `APMINJECT` it has a different filename format, `ddapminstall.msm`, AND it uses a different env var `WINDOWS_APMINJECT_MODULE` instead of `WINDOWS_XXX_DRIVER`.

Trying to accommodate those things made it harder to see what was going on, so it felt cleaner to list them out.

For example, as is I think your suggestion will create the URL path as `/ddapminstallinstall-{info['version']}.msm`